src/components/button/button.test.js

```diff
@@ -41,5 +41,102 @@ describe('/components/button', () => {
       const url = await page.url()
       expect(url).toBe(baseUrl + href)
     })
+    describe('debouncing', () => {
+      it('does not block button links', async () => {
+        // Links should take a user to another part of the page or url, so debouncing it wouldnt have any impact.
+        // But we can check that we're not blocking any initial clicks.
+
+        await page.goto(baseUrl + '/components/button/link/preview', { waitUntil: 'load' })
+
+        const href = await page.evaluate(() => document.body.getElementsByTagName('a')[0].getAttribute('href'))
+
+        // we need to start the waitForNavigation() before the keyboard action
+        // so we return a promise that is fulfilled when the navigation and the click is respectively fulfilled
+        // this is somewhat counter intuitive, as we humans expect to act and then wait for something
+        await Promise.all([
+          page.waitForNavigation(),
+          await page.click('a[role="button"]')
+        ])
+
+        const url = await page.url()
+        expect(url).toBe(baseUrl + href)
+      })
+      it('does not trigger on buttons with type=button', async () => {
+        await page.goto(baseUrl + '/components/button/preview', { waitUntil: 'load' })
+
+        // Our examples don't have form wrappers so we need to overwrite it.
+        await page.evaluate(() => {
+          document.querySelector('button').outerHTML = `
+            <form>
+              <button type="button">Save and continue</button>
+            </form>
+          `
+
+          window.__BUTTON_PRESSED_COUNT = 0
+          document.addEventListener('click', () => {
+            window.__BUTTON_PRESSED_COUNT++
+          })
+        })
+
+        await page.click('button')
+        await page.click('button')
+
+        const buttonPressedCount = await page.evaluate(() => window.__BUTTON_PRESSED_COUNT)
+
+        expect(buttonPressedCount).toBe(2)
+      })
+      it('prevents repeat submissions when in a form', async () => {
+        await page.goto(baseUrl + '/components/button/preview', { waitUntil: 'load' })
+
+        // Our examples don't have form wrappers so we need to overwrite it.
+        await page.evaluate(() => {
+          document.querySelector('button').outerHTML = `
+            <form>
+              <button>Save and continue</button>
+            </form>
+          `
+
+          window.__SUBMIT_COUNT = 0
+          document.querySelector('form').addEventListener('submit', event => {
+            window.__SUBMIT_COUNT++
+            // Don't refresh the page, which will destroy the context to test against.
+            event.preventDefault()
+          })
+        })
+
+        await page.click('button')
+        await page.click('button')
+
+        const submitCount = await page.evaluate(() => window.__SUBMIT_COUNT)
+
+        expect(submitCount).toBe(1)
+      })
+      it('when a user clicks again intentionally it is not prevented', async () => {
```

Comment:
Again to make the intent really clear here (what does intentionally mean?) perhaps something like '(it) does not prevent additional clicks after one second'?